todoman/model.py

```diff
@@ -250,6 +250,27 @@ def complete(self) -> None:
         self.percent_complete = 100
         self.status = "COMPLETED"
 
+    def undo(self) -> Todo | None:
+        """Immediately restores this todo.
+
+        Marks it as needs action, resets the percentage to 0 and deletes the
+        completed_at datetime.
+        If this todo belongs to a series, the one created at completion will
+        be deleted.
+
+        Returns the todo that should be deleted.
+        """
+        recurred = None
+        if self.is_recurring and self.related:
```

Comment:
I ran:

```sh
todo new -l test test
todo new 1
```

The completed todo looks like this:

```icalendar
BEGIN:VCALENDAR
VERSION:2.0
PRODID:io.barrera.todoman
BEGIN:VTODO
COMPLETED:20231118T090119Z
CREATED:20231118T090107Z
DTSTAMP:20231118T090107Z
DUE:20231119T090107Z
LAST-MODIFIED:20231118T090119Z
PERCENT-COMPLETE:100
SEQUENCE:2
STATUS:COMPLETED
SUMMARY:test
UID:1d54c495739f4546a92974ab574b5347@destiny.whynothugo.nl
END:VTODO
END:VCALENDAR
```

It doesn't have a recurrence rule any more (that is moved to the next one in the series), so this condition does not match.

If I run `todo undo 1`, both todos are kept around.

This is not ideal, but I don't think that it is terrible either. The RFC doesn't even cover how a recurring TODO is completed, so there's no correct behavior to any of this either way.

However, this `if` will never run, so this code is misleading. I think it should be removed (unless I'm missing something here).